src/Common/Services/EnvironmentDeploymentHandler.cs

```diff
@@ -78,6 +106,75 @@ private async Task ProcessEnvironmentDeployment(CustomerEnvironment environment,
             }
         }
 
+        private async Task UpdateQueuePositionAsync(long deployableId, CancellationToken token)
+        {
+            bool presentingLoading = false;
+            int? position = null;
+            int initialTopLine;
+            string msg;
+
+            while (!this._isDeploymentFinished && !_hasDeploymentStarted && !token.IsCancellationRequested)
+            {
+                try
+                {
+                    position = new GetMessagePositionInDeploymentQueueInput()
+                    {
+                        DeployableId = deployableId
+                    }.GetMessagePositionInDeploymentQueueSync().MessageQueuePosition;
+
+                    await semaphore.WaitAsync(token);
+
+                    if (position != null && queuePositionCursorCoordinates != null)
+                    {
+                        initialTopLine = Console.CursorTop;
+                        msg = $"{queuePositionMsg} {position}";
+                        Console.SetCursorPosition(queuePositionCursorCoordinates.Value.left, queuePositionCursorCoordinates.Value.top - 1);
+                        _session.LogInformation(msg);
+
+                        queuePositionLoadingCursorCoordinates = (queuePositionCursorCoordinates.Value.left + msg.Length, queuePositionCursorCoordinates.Value.top - 1);
+
+                        if (!presentingLoading)
+                        {
+                            presentingLoading = true;
+                            _ = Task.Run(() => ShowLoadingIndicator(token));
+                        }
+                        Console.SetCursorPosition(0, initialTopLine);
+                    }
+                }
+                catch { }
+                finally
+                {
+                    semaphore.Release();
+                }
+
+                await Task.Delay(5000);
+            }
+        }
+
+        private async Task ShowLoadingIndicator(CancellationToken token)

```

Comment:
This function is duplicated in both files. We may change it to a util file

Reply:
yes it is duplicated but in this method we have a global variable called queuePositionLoadingCursorCoordinates that is shared and used between other methods. And in this method we are in a while, so the method will not return until the cancellation token is received, so its not easy to move this from here (to a shared place)... can we maintain like it is? or do you have suggestions?